test/gurka/test_search_filter.cc

```diff
@@ -182,3 +185,388 @@ TEST_F(SearchFilter, ExcludeRamp) {
+  // exclude_closures set to true (default)
+  {
+    LiveTrafficCustomize close_edge = [](baldr::GraphReader& reader, baldr::TrafficTile& tile,
+                                         int index, baldr::TrafficSpeed* current) -> void {
+      close_single_edge(reader, tile, index, current, "CD", closure_map);
+    };
+    test::customize_live_traffic_data(closure_map.config, close_edge);
+
+    auto result = gurka::route(closure_map, {"1", "2"}, costing, {{"/date_time/type", "0"}}, reader);
+    gurka::assert::osrm::expect_steps(result, {"AB", "CFGD", "DE"});
+    gurka::assert::raw::expect_path(result, {"AB", "BC", "CFGD", "DE"});
+
+    // set ignore_closures in costing, while leaving exclude_closures unset
+    // (which defaults to true). ignore_closures should override
+    // exclude_closures
+    const std::string& req_disable_exclude_closures =
+        (boost::format(
+             R"({"locations":[{"lat":%s,"lon":%s},{"lat":%s,"lon":%s}],"costing":"%s", "costing_options": {"%s": {"speed_types":["freeflow","constrained","predicted","current"], "ignore_closures": true}}, "date_time":{"type":"0"}})") %
+         std::to_string(closure_map.nodes.at("1").lat()) %
+         std::to_string(closure_map.nodes.at("1").lng()) %
+         std::to_string(closure_map.nodes.at("2").lat()) %
+         std::to_string(closure_map.nodes.at("2").lng()) % costing % costing)
+            .str();
+    result = gurka::route(closure_map, req_disable_exclude_closures, reader);
+    gurka::assert::osrm::expect_steps(result, {"AB"});
+    gurka::assert::raw::expect_path(result, {"AB", "BC", "CD", "DE"});
+  }
+}
+
+TEST_P(ExcludeClosuresOnWaypoints, ConsecutiveClosuresAtDeparture) {
+  std::string costing = GetParam();
+
+  // None of the edges are closed. Route has multiple waypoints
+  {
+    auto result = gurka::route(closure_map, {"1", "2"}, costing, {{"/date_time/type", "0"}}, reader);
+    gurka::assert::osrm::expect_steps(result, {"AB"});
+    gurka::assert::raw::expect_path(result, {"AB", "BC", "CD", "DE"});
+  }
+
+  // Close departure & destination edges
+  {
+    LiveTrafficCustomize close_edge = [](baldr::GraphReader& reader, baldr::TrafficTile& tile,
+                                         int index, baldr::TrafficSpeed* current) -> void {
+      close_single_edge(reader, tile, index, current, "AB", closure_map);
+      close_single_edge(reader, tile, index, current, "BC", closure_map);
+      close_single_edge(reader, tile, index, current, "DE", closure_map);
+    };
+    test::customize_live_traffic_data(closure_map.config, close_edge);
+
+    auto result = gurka::route(closure_map, {"1", "2"}, costing, {{"/date_time/type", "0"}}, reader);
+    gurka::assert::osrm::expect_steps(result, {"HIC", "CFGD"});
+    gurka::assert::raw::expect_path(result, {"HIC", "CFGD"});
+
+    // Specify search filter to disable exclude_closures at departure. Due to
+    // consecutive closures, this should result in a "no route"
```

Comment:
I think this test captures the multi-edge closure case, which we do want to return a valid route.

Reply:
For completeness we should also try a test that has closures not just at the origin and destination but also on edge in between, to ensure we allow closures at the endpoints but still avoid them along the route.

Reply:
> For completeness we should also try a test that has closures not just at the origin and destination but also on edge in between, to ensure we allow closures at the endpoints but still avoid them along the route.

Intermediate closure scenario is covered in `IgnoreClosuresOverridesExcludeClosures` testcase. We route from `1 -> 2` and close edge `CD`. Without `ignore_closures` costing option we don't route through CD which is an intermediate edge.

Would you rather see this a separate standalone test (and named appropriately)?

Reply:
I was thinking of adding a new test case that closes the origin edge, destination edge, _and_ an intermediate edge, to exercise the case where a user can snap to a closure but avoid closures in between the snapped edges. Maybe this logic path is covered already though?

Reply:
> I think this test captures the multi-edge closure case, which we do want to return a valid route.

That would depend on whether there are suitable edges nearby, no? If loki's search results in only 1 candidate edge, then I guess if there are no reachable edges to it it would fail to find a route? CMIIW.

Reply:
But in this test the request has `exclude_closures=false`, so we know that there are suitable edges to snap to. Why then does it return a NoRoute?